src/domain/common/usecase/DecryptUrlUseCase.ts

```diff
@@ -0,0 +1,38 @@
+import EthCrypto from 'eth-crypto';
+
+import InvalidEncryptionError from '@core/domain/common/utils/error/InvalidEncryptionError';
+
+class DecryptUrlUseCase {
+  private typeRegex = new RegExp(/(encrypted\(.*?\))/g);
+
+  constructor(private readonly privateKey: string) {
+    this.privateKey = privateKey;
+  }
+
+  async decrypt(url: string): Promise<string> {
+    let result = url;
+
+    if (this.typeRegex.test(url)) {
```

Comment:
reverse if

Reply:
i'm not sure what do you mean. Can you elaborate, please?

Reply:
Agreed face2face.